File: ide-features-trainer/src/training/lesson/LessonLog.java
package training.lesson;

import com.intellij.openapi.util.Pair;

import java.util.ArrayList;
import java.util.Date;

/**
 * Created by karashevich on 13/10/15.
 */
public class LessonLog {

    final private Lesson myLesson;
    private ArrayList<Pair<Date, String>> logData;
    public int exerciseCount = 0;

    public LessonLog(Lesson lesson) {
        myLesson = lesson;
        logData = new ArrayList<Pair<Date, String>>();
        log("Log is created. Lesson:" + lesson.getName());
    }

    public void log(String actionString){
        logData.add(Pair.create(new Date(), actionString));
    }

    public void print(){
        for (Pair<Date, String> dateStringPair : logData) {
            System.out.println(dateStringPair.first + ": " + dateStringPair.first);
        }
    }

    public String exportToString(){
        StringBuilder sb = new StringBuilder();
        for (Pair<Date, String> dateStringPair : logData) {
            sb.append(dateStringPair.first).append(": ").append(dateStringPair.first).append("; ");
        }
        return sb.toString();
    }
}

File: ide-features-trainer/src/training/learn/log/LessonLog.kt
/*
 * Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 */
package training.learn.log

import training.learn.interfaces.Lesson
import java.util.*

class LessonLog(lesson: Lesson) {

  val logData = ArrayList<Pair<Date, String>>()
  var exerciseCount = 0

  fun log(actionString: String) {
    logData.add(Pair(Date(), actionString))
  }

  fun resetCounter() {
    exerciseCount = 0
  }

  fun print() {
    for ((first) in logData) {
      println(first.toString() + ": " + first)
    }
  }

  init {
    log("Log is created. XmlLesson:" + lesson.name)
  }

}
